src/background/permissions/permissions-ensurer.ts

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { BrowserAdapter } from 'common/browser-adapters/browser-adapter';
+
+export class PermissionsEnsurer {
```

Comment:
from @waabid Not sure how I feel about the name: why can't we go with PermissionGetter?

Ensuring we have permissions feels similar to getting permissions to me from a consumer perspective and ensurer seems like an uncommon term for this kind of situation?

Reply:
(from other thread): I'm not a fan of `PermissionGetter`, I don't think it's clear enough that it can be taking a user-visible action rather than just returning a bit of state. My favorite is `BrowserPermissionsRequester`, but I think `Ensurer` is also an okay noun (I think "ensure" is a reasonably common term for this sort of operation)

Reply:
I get what you mean: I like requester!